### PR TITLE
Add specs for `Array#pack`'s `buffer` option

### PR DESCRIPTION
**Pull request by andrykonchin**
Related issue https://github.com/ruby/spec/issues/473

### CODE REVIEW

**Comment by znz** [comment]
How about longer buffer example?

```
[65, 66, 67].pack("@3ccc", buffer: "1234567") #=> "123ABC"
```

**Comment by andrykonchin** [comment]
@znz done

File: core/bignum/abs_spec.rb [hunk 1]
```diff
@@ -1,7 +0,0 @@
-require File.expand_path('../../../spec_helper', __FILE__)
```
Comment (eregon) [inline core/bignum/abs_spec.rb]: I think your commit cc6dc2b2db8aa4abdf1eaaaca95d707bbd0129f0 is inadvertently part of this PR.
Could you remove it?
(One way is `git rebase -i HEAD~3`, or new branch + cherry pick the other commits + reset this branch to the new branch and finally `git push -f`.)

File: core/array/pack/buffer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+require File.expand_path('../../../../spec_helper', __FILE__)
+
+ruby_version_is '2.4' do
+  describe "Aray#pack with `buffer` option" do
+    it "returns specified buffer" do
+      n = [ 65, 66, 67 ]
+      buffer = " "*3
+      result = n.pack("ccc", buffer: buffer)      #=> "ABC"
+      result.object_id.should == buffer.object_id
```
Comment (eregon) [inline core/array/pack/buffer_spec.rb]: There is a built-in matcher to compare identity: equal which calls `equal?`.
`result.should equal buffer`

Comment (andrykonchin) [reply core/array/pack/buffer_spec.rb]: @eregon done

**Pull request merged**